web-local/src/lib/components/navigation/sources/refreshSource.ts

```diff
@@ -1,55 +1,71 @@
-import type { V1PutFileAndReconcileResponse } from "@rilldata/web-common/runtime-client";
-import { config } from "@rilldata/web-local/lib/application-state-stores/application-store";
+import type {
+  V1PutFileAndReconcileResponse,
+  V1RefreshAndReconcileResponse,
+} from "@rilldata/web-common/runtime-client";
 import { fileArtifactsStore } from "@rilldata/web-local/lib/application-state-stores/file-artifacts-store";
 import { overlay } from "@rilldata/web-local/lib/application-state-stores/overlay-store";
 import { compileCreateSourceYAML } from "@rilldata/web-local/lib/components/navigation/sources/sourceUtils";
 import {
   openFileUploadDialog,
   uploadFile,
 } from "@rilldata/web-local/lib/util/file-upload";
-import type { UseMutationResult } from "@sveltestack/svelte-query";
+import type { QueryClient, UseMutationResult } from "@sveltestack/svelte-query";
+import { EntityType } from "../../../../common/data-modeler-state-service/entity-state-service/EntityStateService";
+import {
+  invalidateAfterReconcile,
+  invalidateTablewideProfilingQueries,
+} from "../../../svelte-query/invalidation";
+import { getFileFromName } from "../../../util/entity-mappers";
 
 export async function refreshSource(
   connector: string,
-  tableName: string,
+  sourceName: string,
   instanceId: string,
-  refreshSource: UseMutationResult,
-  createSource: UseMutationResult<V1PutFileAndReconcileResponse>
+  refreshSource: UseMutationResult<V1RefreshAndReconcileResponse>,
+  createSource: UseMutationResult<V1PutFileAndReconcileResponse>,
+  queryClient: QueryClient
 ) {
-  if (connector === "file") {
-    const files = await openFileUploadDialog(false);
-    if (!files.length) return Promise.reject();
-
-    overlay.set({ title: `Importing ${tableName}` });
-    const filePath = await uploadFile(
-      `${config.database.runtimeUrl}/v1/instances/${instanceId}/files/upload`,
-      files[0]
-    );
-    if (filePath) {
-      const yaml = compileCreateSourceYAML(
-        {
-          sourceName: tableName,
-          path: filePath,
-        },
-        "file"
-      );
-      const resp = await createSource.mutateAsync({
+  if (connector !== "file") {
+    overlay.set({ title: `Importing ${sourceName}` });
+    const resp = await refreshSource.mutateAsync({
+      data: {
         instanceId,
-        data: {
-          instanceId,
-          path: `sources/${tableName}.yaml`,
-          blob: yaml,
-          create: true,
-          strict: true,
-        },
-      });
-      fileArtifactsStore.setErrors(resp.affectedPaths, resp.errors);
-    }
-  } else {
-    overlay.set({ title: `Importing ${tableName}` });
-    await refreshSource.mutateAsync({
-      instanceId,
-      name: tableName,
+        path: `sources/${sourceName}.yaml`,
+      },
     });
+    invalidateAfterReconcile(queryClient, instanceId, resp);
+    invalidateTablewideProfilingQueries(queryClient, instanceId, sourceName);
```

Comment:
`invalidateTablewideProfilingQueries` should go in `invalidateAfterReconcile`. Add a condition there based on type of artifact. Because this might be needed for models as well later on. So adding in the method which is used everywhere makes more sense.